### DISCUSSION TITLE
Questions for CLV customer_lifetime_value params and clv_estimate quantile selection

### DISCUSSION THREAD

**Discussion by BeiYaaa** [pymc-labs/pymc-marketing#584] [Q&A] [answered]
Hi there,
I'm quite confused by the parameters of the function `customer_lifetime_value`. Hope anyone could help here.

The rfm data I have is on hourly level, so I set up the `BetaGeo `and `GammaGamma `models with the `frequency`, `monetary_value` etc based on hours. However, when to use `expected_customer_lifetime_value `function, I noticed that the time is defined by months.
```
time: int, optional
        The lifetime expected for the user in months. Default: 12
```
And for `freq `which is an additional parameter, Unite of time needs to be specified again. My understanding is that different levels are designed to be converted to iterate in the same way.
```
freq: string, optional
        Unit of time of the purchase history. Defaults to "D" for daily.
        Other options are "W" (weekly), "M" (monthly), and "H" (hourly).
        Example: If your dataset contains information about weekly purchases,
        you should use "W".
```

According to this, I coded my script as shown below. I wanted to see the LTV for each customers for the future one month. `t` is defined as 30*24.
```
clv_estimate = gg.expected_customer_lifetime_value(
                        transaction_model=bgm,
                        customer_id=selected_train_df_gg['customer_id'],
                        mean_transaction_value=selected_train_df_gg["monetary_value"],
                        frequency=selected_train_df_gg["frequency"],
                        recency=selected_train_df_gg["recency"],
                        T=selected_train_df_gg["T"],
                        time=t/24, 
                        freq='H',
                        discount_rate=0.01, # monthly discount rate ~ 12.7% annually
                    )
```

However, when I compare the expected values (y axis) vs. original one (x aixs), it looks that the model is very over-predicting. I wonder if I set the parameters correctly.
![image](https://github.com/pymc-labs/pymc-marketing/assets/33733796/363eca66-156a-4c1c-9cd0-8eff19efc53c)

Additional codes:
```
actual_ltv_df = test_df.groupby('u_id').sum(numeric_only=1).reset_index()
# make actual vs modeled exp_clv df
avm = pd.merge(actual_ltv_df,
                pd.DataFrame({'customer_id': selected_train_df_gg['customer_id'],
                                'exp_clv': clv_estimate.quantile(0.5, dim=("chain", "draw")).values}),
                how='right',
                right_on='customer_id',
                left_on='u_id')
sns.jointplot(avm,
                    x='actual', 
                    y='exp_clv',
                    alpha=0.1)
```

Besides, I wonder if anyone could share some knowledge regarding the clv_estimate quantile. Have you chosen other quantile instead of mean for the outcome. If you have, what are the considerations?

Thanks in advance!

**Comment by ColtAllen** [comment, marked as answer]
Hey @BeiYaaa,

The way the code is currently written, CLV is being estimated for 30 months. You can set `time=1` if you're only interested in one month, and `freq='H'` will handle the time conversion.

To answer your second question, a quantile of 0.5 is actually the median. It's worth comparing against the mean (a large discrepancy usually indicates outliers) but the question of which is best depends on your use case.
